### ISSUE TITLE
Use of unsupported Linq operations on Unity AOT (iOS) platforms

### ISSUE THREAD

**Issue by manish-mathai** [aws/aws-sdk-unity#40]
Certain Linq operations, like OrderBy, require JIT compilation to run. Unity3D AOT platforms like iOS do not support JIT and these operations fail at runtime with the error such as 
```
ExecutionEngineException: Attempting to JIT compile method 'System.Linq.OrderedEnumerable`1<System.Collections.Generic.KeyValuePair`2<string, string>>:GetEnumerator ()' while running with --aot-only.
```
An example of such use is in Amazon.Runtime.Internal.Auth.AWS4Signer.CanonicalizeHeaders and other related methods. 

Possible solutions would be to either reduce reliance on unsupported operations or use AOT-supported Linq implementations like [UniLinq](https://github.com/RyotaMurohoshi/UniLinq)

**Comment by karthiksaligrama** [comment]
Thanks for reporting. We are aware of this issue and are working for a fix.

**Comment by karthiksaligrama** [comment]
Fixed in version 1.0.5. Please update the sdk and let us know if it fixes the problem
